sdk/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Util.hs

```diff
@@ -459,6 +459,39 @@ instance Show UpgradingDep where
       Just udPackageVersion -> " (v" <> show udPackageVersion <> ")"
       Nothing -> mempty
 
+------------------------------------------------------------------------
+-- Shorthands
+------------------------------------------------------------------------
+
+-- Types
+tvar :: T.Text -> Type
+tvar = TVar . TypeVarName
+
+tunit :: Type
+tunit = TBuiltin BTUnit
+
+tint :: Type
+tint = TBuiltin BTInt64
+
+tbool :: Type
+tbool = TBuiltin BTBool
+
+tyLamTyp :: Type
+tyLamTyp = TForall (a, typToTyp) (TVar a :-> TVar a)
+  where
+    a = TypeVarName "a"
+    typToTyp = KArrow KStar KStar
+
+tcon :: T.Text -> Type
+tcon t = TCon $ Qualified SelfPackageId (ModuleName ["Main"]) (TypeConName [t])
+
+tsyn :: T.Text -> [Type] -> Type
+tsyn t = TSynApp $ Qualified SelfPackageId (ModuleName ["Main"]) (TypeSynName [t])
+
+tmaybe :: Type -> Type
+tmaybe = TApp (tcon "Maybe")
```

Comment:
We don't have a `Maybe` in LF so much as we have `TBuiltin BTOptional`, I'd suggest you create a custom tcon "MyFunctor" inside the encode/decode tests instead.